app/assets/javascripts/3dbio_viewer/src/data/repositories/BionotesPdbInfoRepository.ts

```diff
@@ -54,7 +85,28 @@ export class BionotesPdbInfoRepository implements PdbInfoRepository {
     }
 }
 
-type UniprotFromPdbMapping = Record<PdbId, Record<ProteinId, ChainId[]>>;
+type ErrorType = "serviceUnavailable" | "noData";
+
+function throwError<T>(type: ErrorType): FutureData<T> {
+    switch (type) {
+        case "serviceUnavailable": return Future.error({
+            message: i18n.t(
+                "We apologize. Some of the services we relay on, are temporarily unavailable. Our team is working to resolve the issue, and we appreciate your patience. Please try again later."
+            ),
+        });
+        case "noData": return Future.error({
+            message: i18n.t(`No data found for this PDB. But you can try and visualize another PDB. If you believe this is incorrect, please contact us using the "Send Feedback" button below.`),
+        })
+    }
+}
+
+type UniprotFromPdbMapping = Record<PdbId, Record<ProteinId, ChainId[]> | never[]>;
```

Comment:
It's usually an object but when the entry doesn't have Uniprot mapping for the entry, it returns an empty array. That's why I put the `never[]`. Indeed instead of throwing error when the entry is not found, it just outputs successfully with no response. You can check: [6zow (expected)](http://server:3001/3dbionotes/api/mappings/PDB/Uniprot/6zow), [8uys (no uniprot)](http://server:3001/3dbionotes/api/mappings/PDB/Uniprot/8uys), [9zzz (entry doesn't exist)](http://server:3001/3dbionotes/api/mappings/PDB/Uniprot/9zzz)